tikv-client/src/main/java/com/pingcap/tikv/key/TypedKey.java

```diff
@@ -72,14 +82,23 @@ public static TypedKey toTypedKey(Object val, DataType type, int prefixLength) {
    * @return next TypedKey with same prefix length
    */
   public TypedKey next() {
+    DataType tp = getType();
     Object val = getValue();
-    if (val instanceof String) {
+    if (tp instanceof StringType) {
       return toTypedKey(prefixNext(((String) val).getBytes()), type, prefixLength);
-    } else if (val instanceof byte[]) {
+    } else if (tp instanceof BytesType) {
       return toTypedKey(prefixNext(((byte[]) val)), type, prefixLength);
+    } else if (prefixLength == -1) {
```

Comment:
if the prefix length is undefined, we better encapsulate it into function such as `isPrefixUndefined` ?

Reply:
sure

Reply:
done.